tests/Database/DatabaseEloquentWithPropertyHooksTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Illuminate\Tests\Database;
+
+use Illuminate\Database\Capsule\Manager as DB;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Tests\Database\Fixtures\Models\EloquentModelWithAllPropertyHooks;
+use Illuminate\Tests\Database\Fixtures\Models\EloquentModelWithAppendedPropertyHooks;
+use Illuminate\Tests\Database\Fixtures\Models\EloquentModelWithGetterPropertyHook;
+use Illuminate\Tests\Database\Fixtures\Models\EloquentModelWithPrecedentPropertyHook;
+use Illuminate\Tests\Database\Fixtures\Models\EloquentModelWithPropertyHooks;
+use Illuminate\Tests\Database\Fixtures\Models\EloquentModelWithSetterPropertyHook;
+use PHPUnit\Framework\TestCase;
+
+class DatabaseEloquentWithPropertyHooksTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (PHP_VERSION_ID < 80400) {
+            $this->markTestSkipped(
+                'Property Hooks are not available to test in PHP '.PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION,
+            );
+        }
```

Comment:
You could alternatively use [PHPUnit's attribute annotations](https://docs.phpunit.de/en/10.5/attributes.html#requiresphp) to achieve this if you like:
```suggestion
#[RequiresPhp('8.4.0')]
class DatabaseEloquentWithPropertyHooksTest extends TestCase
{
    protected function setUp(): void
    {
        parent::setUp();
```